src/PlugKeyRing/index.ts

```diff
@@ -292,6 +292,15 @@ class PlugKeyRing {
 
     const { [walletId]: deletedWallet, ...maintainedWallets } = wallets
 
+    if (walletId == this.currentWalletId) {
+      const mainWalletId = this.getMainAccountId()
+
+      const currentWalletId = maintainedWallets[mainWalletId].walletId;
```

Comment:
mainWalletId and *.walletId is not the same? May be could be easier just make 
`currentWalletId = mainWalletId`